src/document/DocumentAnnotator.ts

```diff
@@ -66,7 +66,24 @@ export default class DocumentAnnotator extends BaseAnnotator {
             if (this.isFeatureEnabled('highlightText')) {
                 managers.add(new HighlightManager({ location: pageNumber, referenceEl: pageReferenceEl }));
             }
-            managers.add(new RegionManager({ location: pageNumber, referenceEl: pageReferenceEl }));
+            if (this.isFeatureEnabled('discoverability')) {
+                const canvasLayerEl = pageEl.querySelector('.canvasWrapper') as HTMLElement;
+                const referenceEl = canvasLayerEl || pageReferenceEl;
```

Comment:
Does it still make sense to have a central `this.getPageReference` method if the reference element is now different per-annotation type?

Reply:
From what I see it looks like `this.getPageReference` acts as a sort of catch-all method in which we just try to return the canvas / text / region layer if they exist. I think we still need it as the default method that we rely on and then later if we need an annotation-specific type we apply it over the default.